### PR TITLE
[SPARK-30684][WEBUI] Show the descripton of metrics for WholeStageCodegen in DAG viz

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

Added description for metrics shown in the WholeStageCodegen-node in DAG viz.

This is before the change is applied.
before-changed

And following is after change.
after-fixing-layout

For this change, I also modify the layout of DAG viz.
Actually, I noticed  it's not enough to just added the description.
Following is without changing the layout.
layout-is-broken

### Why are the changes needed?

Users can't understand what those metrics mean.

### Does this PR introduce any user-facing change?

Yes. The layout is a little bit changed. 

### How was this patch tested?

I confirm the result of DAG viz with following 3 operations.

`sc.parallelize(1 to 10).toDF.sort("value").filter("value > 1").selectExpr("value * 2").show`
`sc.parallelize(1 to 10).toDF.sort("value").filter("value > 1").selectExpr("value * 2").write.format("json").mode("overwrite").save("/tmp/test_output")`
`sc.parallelize(1 to 10).toDF.write.format("json").mode("append").save("/tmp/test_output")`

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala [hunk 1]
```diff
@@ -169,12 +169,17 @@ private[ui] class SparkPlanGraphNode(
       metric.name + ": " + value
     }
 
+    // If there are metrics, display each entry in a separate line.
+    // Note: whitespace between two "\n"s is to create an empty line between the name of
+    // SparkPlan and metrics. If removing it, it won't display the empty line in UI.
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala]: Thank you for the note.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala [hunk 2]
```diff
@@ -198,7 +203,7 @@ private[ui] class SparkPlanGraphCluster(
       require(duration.length == 1)
       val id = duration(0).accumulatorId
       if (metricsValue.contains(duration(0).accumulatorId)) {
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala]: While you are here, can we change like this?
```scala
- if (metricsValue.contains(duration(0).accumulatorId)) {
+ if (metricsValue.contains(id)) {
```

Comment (sarutak) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala]: Sure. I'll do it.

**Review by dongjoon-hyun — COMMENTED** [review]
@sarutak . I verified that the function is working, but this PR seems to have a negative side-effect. As you describe in the PR description, this PR change the space of the other operator like `Project`.

Screen Shot 2020-01-30 at 10 17 37 AM

Screen Shot 2020-01-30 at 10 18 01 AM

It would be great if we can remove that side effect on the other parts.

**Comment by sarutak** [comment]
@dongjoon-hyun I tried some variation of changes and making the height of boxes higher works. How about centering those labels?

**Comment by dongjoon-hyun** [comment]
+1 for centering~

**Comment by sarutak** [comment]
This is the result of centering labels.
Screenshot from 2020-01-31 03-48-11

**Comment by dongjoon-hyun** [comment]
@sarutak . At this time, `Project` is not **bold**.
Screen Shot 2020-01-30 at 11 11 58 AM

File: sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.css [hunk 3]
```diff
@@ -33,7 +33,7 @@
 }
 
 /* Highlight the SparkPlan node name */
-#plan-viz-graph svg text :first-child {
+#plan-viz-graph svg text :first-child, #plan-viz-graph svg text :nth-child(2) {
```
Comment (dongjoon-hyun) [inline sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.css]: Is there any side-effect with `:nth-child(2)`?

Comment (sarutak) [reply sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.css]: I  concerned about side effects but another idea is a little bit tricky.
I'll submit another one.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @sarutak !
Merged to master for 3.0.0.

**Pull request closed without merging**